yt_dlp/extractor/viki.py

```diff
@@ -295,15 +309,35 @@ def _real_extract(self, url):
                 'id': thumbnail_id,
                 'url': thumbnail.get('url'),
             })
+        
+        stream_url = 'playback_streams/%s.json?drms=dt1,dt2&device_id=%s&token=%s' % (video_id, self._DEVICE_ID, self._token)
+        resp = self._call_api(stream_url, video_id, 'Downloading video streams JSON')
 
+        stream_id = resp['main'][0]['properties']['track']['stream_id']
+        
+        subtitle_format = 'https://api.viki.io/v4/videos/%s/auth_subtitles/%s.%s?app=%s&stream_id=%s'
         subtitles = {}
         for subtitle_lang, _ in (video.get('subtitle_completions') or {}).items():
+
             subtitles[subtitle_lang] = [{
                 'ext': subtitles_format,
-                'url': self._prepare_call(
-                    'videos/%s/subtitles/%s.%s' % (video_id, subtitle_lang, subtitles_format)),
+                'url': subtitle_format % (video_id, subtitle_lang, subtitle_format, self._APP, stream_id)
             } for subtitles_format in ('srt', 'vtt')]
 
+        mpd_url = self.search_another_mpd_url(resp['main'][0]['url'], video_id)
+
+        '''
+        NOTE: audio formats should be sorted by height which can get from audio dash url.
+        '''
```

Comment:
It's just to sort the high audio from the audio url.
because of this:
They have same bitrate. When you sort it, it will choose the id "8" instead of 9.

`<Representation id="8" mimeType="audio/mp4" codecs="mp4a.40.2" audioSamplingRate="44100" startWithSAP="1" bandwidth="194473">
        <AudioChannelConfiguration schemeIdUri="urn:mpeg:dash:23003:3:audio_channel_configuration:2011" value="2"/>
        <BaseURL>1183352v_dash_high_1080p_2a98c0_2106140900_track2_dashinit.mp4</BaseURL>
        <SegmentBase indexRangeExact="true" indexRange="853-10364">
          <Initialization range="0-852"/>
        </SegmentBase>
      </Representation>
      <Representation id="9" mimeType="audio/mp4" codecs="mp4a.40.2" audioSamplingRate="44100" startWithSAP="1" bandwidth="194473">
        <AudioChannelConfiguration schemeIdUri="urn:mpeg:dash:23003:3:audio_channel_configuration:2011" value="2"/>
        <BaseURL>1183352v_dash_high_720p_00c68c_2106140800_track2_dashinit.mp4</BaseURL>
        <SegmentBase indexRangeExact="true" indexRange="853-10364">
          <Initialization range="0-852"/>`